frame/democracy/src/lib.rs

```diff
@@ -647,6 +679,7 @@ pub mod pallet {
 
 			<Cancellations<T>>::insert(h, true);
 			Self::internal_cancel_referendum(ref_index);
+			Self::clear_metadata(MetadataOwner::Referendum(ref_index));
```

Comment:
Why not move this call into the `internal_cancel_referendum` function?
Since they get called together.

Reply:
I did not want to extend the scope of internal_cancel_referendum, and inject_referendum, because they are public.
But now I think I better do, because of the same reason. 
I can not imagine a case when the metadata should not be cleared or reset on cancel and inject new

Reply:
left inject_referendum as it is, it used when there is no prior proposal